Rate limit technical design (#930)
kyma-project/istio: docs/contributor/adr/0008-rate-limit.md

While enabling rate limit feature on the HTTP layer, there is a possibility to give insights to the client about the current rate limiting state e.g. how many requests are left to be done before being rate limited, or how much time is left until rate limit tokens are refilled. 
The decision is that rate limit headers can be enabled by the user in the Rate Limit CR by setting the boolean field `spec.enableHeaders`. Following security good practices the exposure of internal information is limited by disabling those headers by default.

### RateLimit CR Spec

| field                      | type                | description                                                                                                                                                                                                                                               | required |
|----------------------------|---------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|

> | context                    | string              | Allows to set context for the underlying EnvoyFilter. Can be "gateway", which means rate limit will be applied to the Istio Ingress Gateways, or "sidecar" which means it will be applied to sidecar proxies of the workloads.                            | yes      |

Reviewer: ```suggestion
| context                    | string              | Allows to set the context for the underlying EnvoyFilter. The value can be `gateway`, which means the rate limit is applied to the Istio Ingress Gateways, or `sidecar`, which means it is applied to sidecar proxies of the workloads.                            | yes      |
```

> | enableHeaders              | boolean             | Enable x-rate-limit response headers, the default value is false                                                                                                                                                                                          | yes      |

Reviewer: ```suggestion
| enableHeaders              | boolean             | Enables **x-rate-limit** response headers. The default value is `false`.                                                                                                                                                                                         | yes      |
```
